tools/c7n_left/c7n_left/core.py

```diff
@@ -42,17 +42,16 @@ def run(self):
         provider = self.get_provider()
 
         if not provider.match_dir(self.options.source_dir):
-            raise NotImplementedError(
-                "no %s source files found" % provider.provider_name
-            )
+            log.warning("no %s source files found" % provider.type)
+            return 1
```

Comment:
This looks fine to me, but made me wonder when we return _not_ 1. Looks like below we poke through the resource graph and return 0 (or `False`) only if there were policies and resources, but no resource types matching our policy set. Is that expected? (Fair warning I might be misreading this)

Reply:
for c7n-left we always exit 1 on resource match to a policy, effectively fail ci if non compliant found. In this case we fail because we have no policies which smells like a misconfig. the cli effectively does sys.exit(int(return_value_of_this_method))

Reply:
I guess we can return true for consistency here.

Reply:
added a type annotation for the return value and inline comment wrt to using return value for exit code.